File: .github/smoke/java/ExternalConsumer.java
import dev.oppex.sdk.api.IncidentClient;
import dev.oppex.sdk.model.IncidentRequest;
import dev.oppex.sdk.model.Severity;

/** Compiled from scratch by each CI JDK to verify the supported consumer API. */
public final class ExternalConsumer {
    private ExternalConsumer() {
    }

    public static void main(String[] args) throws Exception {
        Class.forName("org.apache.http.impl.client.CloseableHttpClient");
        Class.forName("com.fasterxml.jackson.core.JsonFactory");

        IncidentClient client = IncidentClient.builder()
                .apiKey("external-consumer-api-key")
                .serviceKey("external-consumer-service-key")
                .build();
        try {
            IncidentRequest request = IncidentRequest.builder()
                    .title("External consumer compilation test")
                    .source("github-actions")
                    .severity(Severity.MEDIUM)
                    .build();
            if (request.getSeverity().getValue() != 3) {
                throw new IllegalStateException("Unexpected severity mapping");
            }
        } finally {
            client.close();
        }

        // Service routing needs no service key, and its precondition fails before any network call.
        IncidentClient routingClient = IncidentClient.builder()
                .apiKey("external-consumer-api-key")
                .build();
        try {
            IncidentRequest keyed = IncidentRequest.builder()
                    .title("Service routing compilation test")
                    .source("github-actions")
                    .severity(Severity.LOW)
                    .serviceKey("external-consumer-service-key")
                    .build();
            try {
                routingClient.postWithServiceRouting(keyed);
                throw new IllegalStateException("Expected a request service key to be rejected");
            } catch (IllegalArgumentException expected) {
                // Service routing must refuse a request that carries its own service key.
            }
        } finally {
            routingClient.close();
        }
        System.out.println("EXTERNAL_CONSUMER_OK java=" + System.getProperty("java.version"));
    }
}

File: .github/smoke/python/external_consumer.py
"""Byte-compiled and executed by each CI interpreter to verify the consumer API.

This file may use only the supported SDK API and Python 2.7-compatible syntax.
"""

from __future__ import absolute_import, print_function

import sys

from oppex_sdk import IncidentClient, IncidentRequest, Severity


def main():
    with IncidentClient(api_key="external-consumer-api-key",
                        service_key="external-consumer-service-key") as client:
        request = IncidentRequest(
            title="External consumer compatibility test",
            source="github-actions",
            severity=Severity.MEDIUM,
        )
        if request.severity.value != 3:
            raise AssertionError("Unexpected severity mapping")
        if client.closed:
            raise AssertionError("A new client must not be closed")

    # Service routing needs no service key, and its precondition fails before any network call.
    with IncidentClient(api_key="external-consumer-api-key") as routing_client:
        keyed = IncidentRequest(
            title="Service routing compatibility test",
            source="github-actions",
            severity=Severity.LOW,
            service_key="external-consumer-service-key",
        )
        try:
            routing_client.post_with_service_routing(keyed)
        except ValueError:
            pass  # Service routing must refuse a request that carries its own service key.
        else:
            raise AssertionError("Expected a request service key to be rejected")

    print("EXTERNAL_CONSUMER_OK python=" + sys.version.split()[0])
    return 0


if __name__ == "__main__":
    sys.exit(main())
